lib/proto/http.js

```diff
@@ -27,14 +27,13 @@ module.exports = {
         let fallbackRetryDelayInMs = ms(opts.fallbackRetryDelay || '60s');
 
         const options = {
-            uri: encodeURI(link),
             headers: {
                 // override 'User-Agent' (some sites return `401` when the user-agent isn't a web browser)
                 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36'
             },
-            maxRedirects: 8,
-            strictSSL: false,
-            timeout: ms(timeout),
+            follow_max: 8,
+            rejectUnauthorized: false,
```

Comment:
I copied that in my PR but then reading the doc a bit more I realised this was just an option passed to the Node request directly https://nodejs.org/docs/latest/api/tls.html#tls_tls_connect_options_callback

But as indicated in the doc https://nodejs.org/docs/latest/api/tls.html#tls_tls_createserver_options_secureconnectionlistener , this has no effect if requestCert is false (and it's false by default).
> rejectUnauthorized <boolean> If not false the server will reject any connection which is not authorized with the list of supplied CAs. **This option only has an effect if requestCert is true**. Default: true.

But also it's the `tls.createServer` doc so, I wonder if it exactly works the same for requests. Do you have some experience with this? Could we add a test with a self signed cert to make sure it works?

Reply:
Apologies, I've been a bit overwhelmed with some changes :)
I'm definitely with working on one PR as a starter.
I can dig into the above once I am a bit free, and I'll review the other PR and help.

Reply:
Please, know that I totally understand how hard it is to deal with all the stuff we would like to...
I suggest we close this one for clarity and then we both work together on the other one.
Would that be fine for you?

Reply:
Go ahead!